### PR TITLE
[tests-only][full-ci] Improve Then step in apiShareReshareToShares3 suite

### PR DESCRIPTION
**Pull request by amrita-shrestha**
## Description
This PR adds `Then` steps to existing API test scenarios to better check the results of the `When` steps.

**Suites covered:**
- `improveThenStepInApiShareReshareToShare3`

## Related Issue

- Part of https://github.com/owncloud/core/issues/39512

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

File: tests/acceptance/features/apiShareReshareToShares3/reShareWithExpiryDate.feature [hunk 1]
```diff
@@ -430,6 +406,7 @@ Feature: resharing a resource with an expiration date
     #The action of changing the expiration date while resharing should be forbidden
     Then the HTTP status code should be "403"
     And the OCS status code should be "403"
+    And the sharing API should report to user "Carol" that no shares are in the pending state
```
Comment (SwikritiT) [inline tests/acceptance/features/apiShareReshareToShares3/reShareWithExpiryDate.feature:409]: I'm sorry but what is the need of this step here? if both the HTTP and OCS status code are `403`
Imo the scenario name `reshare can extend the received expiry date further into the future`  doesn't match what the scenario is doing. I don't see any extension of the expiry date after resharing.

Comment (amrita-shrestha) [reply tests/acceptance/features/apiShareReshareToShares3/reShareWithExpiryDate.feature:409]: I add this test to make sure Carol doesn't have any share requests

**Review by SagarGi — COMMENTED** [review]
lgtm

**Review by SwikritiT — COMMENTED** [review]
` enforce-expire-date` is set to `yes` for all so maybe we don't need to put it in example table, we can put that directly in step in line 67. Similar for other example tables too
https://github.com/owncloud/core/blob/3675d80373fd6045d6d8d2991857f9d5e6c87016/tests/acceptance/features/apiShareReshareToShares3/reShareWithExpiryDate.feature#L84-L89

File: tests/acceptance/features/apiShareReshareToShares3/reShareWithExpiryDate.feature [hunk 2]
```diff
@@ -326,25 +308,23 @@ Feature: resharing a resource with an expiration date
...
-      | ocs_api_version | default-expire-date | enforce-expire-date | expected-expire-date | ocs_status_code |
-      | 1               | yes                 | yes                 | +20 days             | 100             |
-      | 2               | yes                 | yes                 | +20 days             | 200             |
-      | 1               | no                  | yes                 | +20 days             | 100             |
-      | 2               | no                  | yes                 | +20 days             | 200             |
+      | ocs_api_version | default-expire-date | expected-expire-date | ocs_status_code |
+      | 1               | yes                 | +20 days             | 100             |
+      | 2               | yes                 | +20 days             | 200             |
+      | 1               | no                  | +20 days             | 100             |
+      | 2               | no                  | +20 days             | 200             |
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiShareReshareToShares3/reShareWithExpiryDate.feature:321]: `expected-expire-date` column also has the same values

**Review by SwikritiT — COMMENTED** [review]
LGTM

**Review by phil-davis — APPROVED** [review]

**Pull request merged**